server/data_server.c

```diff
@@ -191,6 +194,10 @@ static error__t start_data_capture(void)
 error__t arm_capture(void)
 {
     unsigned int readers, active;
+
+    // get PCAP ARM timestamp and store as a global
+    clock_gettime(CLOCK_REALTIME, &pcap_arm_ts);
```

Comment:
It is highly improbably that `clock_gettime` will fail ... but we have an error handling chain on the line directly below, so move this down to just below the `return` on the next line and write this as
```C
return
    TEST_IO(clock_gettime(CLOCK_REALTIME, &pcap_arm_ts))  ?:
    TEST_OK_(check_pcap_valid(), ...
```

This means that if for whatever reason this call _did_ fail we'd get an error report.

Reply:
Ah.  I've realised the issue is a bit more tricky: you have to move the `clock_gettime()` call inside the lock and after the test that `data_capture_enabled` isn't already set and clients aren't still taking data: if you don't do this, then someone pressing `ARM` during a capture will reset the timestamp!

Rewrite this function thus:
```C
return
    TEST_OK_(check_pcap_valid(),
        "PCAP not supported with this configuration")  ?:
    WITH_LOCK(data_thread_mutex,
        TEST_OK_(!data_capture_enabled,
            "Data capture already in progress")  ?:
        /* If data capture is not enabled then we can safely expect the
         * buffer status to be idle. */
        TEST_OK(!read_buffer_status(data_buffer, &readers, &active))  ?:
        TEST_OK_(active == 0, "Data clients still taking data")  ?:
        /* Get PCAP ARM timestamp and store as a global */
        TEST_IO(clock_gettime(CLOCK_REALTIME, &pcap_arm_ts))  ?:
        start_data_capture());
```